granulate_utils/metadata/databricks_client.py

```diff
@@ -135,12 +129,54 @@ def _cluster_all_tags_metadata(self) -> Optional[Dict[str, str]]:
         props = env.get("sparkProperties")
         if props is None:
             raise DatabricksJobNameDiscoverException(f"sparkProperties was not found in {env=}")
-        for prop in props:
-            if prop[0] == CLUSTER_TAGS_KEY:
-                try:
-                    all_tags_value = json.loads(prop[1])
-                except Exception as e:
-                    raise DatabricksJobNameDiscoverException(f"Failed to parse {prop=}") from e
-                return {cluster_all_tag["key"]: cluster_all_tag["value"] for cluster_all_tag in all_tags_value}
+        # Creating a dict of the relevant properties and their values.
+        service_name_prop_candidates = {
+            prop[0]: prop[1] for prop in props if (CLUSTER_ALL_TAGS_PROP == prop[0] or CLUSTER_NAME_PROP == prop[0])
+        }
+        if len(service_name_prop_candidates) == 0:
+            # We expect at least one of the properties to be present.
+            raise DatabricksJobNameDiscoverException(f"Failed to create dict of relevant properties {env=}")
+        # First, trying to extract `CLUSTER_TAGS_KEY` property, in case not redacted.
+        if (
+            cluster_all_tags_value := service_name_prop_candidates.get(CLUSTER_ALL_TAGS_PROP)
+        ) is not None and "redacted" not in cluster_all_tags_value:
+            try:
+                cluster_all_tags_value_json = json.loads(cluster_all_tags_value)
+            except Exception as e:
+                raise DatabricksJobNameDiscoverException(f"Failed to parse {cluster_all_tags_value}") from e
+            return self._apply_pattern(
+                {cluster_all_tag["key"]: cluster_all_tag["value"] for cluster_all_tag in cluster_all_tags_value_json}
+            )
+        # As a fallback, trying to extract `CLUSTER_NAME_PROP` property.
+        elif (cluster_name_value := service_name_prop_candidates.get(CLUSTER_NAME_PROP)) is not None:
+            return self._apply_pattern({CLUSTER_NAME_KEY: cluster_name_value})
```

Comment:
Is this a key that might be present in `cluster_all_tags_value_json` ? If not, then what about having a different `_apply_pattern` for `JOB_NAME_KEY` and another for `CLUSTER_NAME_KEY`?

Reply:
yes, `CLUSTER_NAME_KEY` can appear in both